### DISCUSSION TITLE
Recreating sym links and small,medium,thumbs

### DISCUSSION THREAD

**Discussion by opravil-jan** [LycheeOrg/Lychee#2582] [General]
Hello,

is there a way to recreating sym links and small, medium and thumbs pictures from original? I went through artisan commands and did not found any option there.

One more question. Why there is small and small2x directory?

Thank you

**Comment by ildyria** [comment]
sym links are automatically recreated. You can set the expiration in the config. Otherwise you can also just empty the symlink table and the `/sym` folder in your Lychee install. The links will be created on the fly when the user access the pictures.

`small2x` contains `small` thumbs but with twice the resolution, those are meant to be used on HiDPI devices e.g. retina display (just like `medium2x` for `medium`). They can be enabled/disabled in the configuration, so it is an opt-in functionality.

In order to regenerate the smaller size variants of an original image there is the `php artisan lychee:generate_thumbs` command.

**Reply by opravil-jan** [reply]
Thank you very much for explanation.

**Reply by opravil-jan** [reply]
Will it be also creating record to database using php artisan lychee:generate_thumbs?

**Reply by ildyria** [reply]
`lychee: generate_thumbs` generates the files and the database entries.
